lib/webmushra/spatial/Scene.js

```diff
@@ -0,0 +1,448 @@
+  if(this.wmCamera.type == 1){ //orthographic camera
+  	if(this.viewConfig.view == 'front'){
+		position[1] = -_relY + this.divHeight /2;
+		position[0] = _relX;
+	} else if(this.viewConfig.view == 'back'){
+		position[1] = -_relY + this.divHeight /2;
+		position[0] = -_relX;
+	} else if(this.viewConfig.view == 'top'){
+		position[0] = _relX;
+		position[2] = _relY;
+	} else if(this.viewConfig.view == 'right'){
+		position[1] = -_relY + this.divHeight /2;
+		position[2] = -_relX ;
+	} else if(this.viewConfig.view == 'left'){
+		position[1] = -_relY + this.divHeight/2;
+		position[2] = _relX ;
+	}
+
+  }else { 
+  	var cameraDir = (new THREE.Vector3(-this.wmCamera.getCameraX(), -this.wmCamera.getCameraY(),-this.wmCamera.getCameraZ()));
+  	var cameraDir_n = new THREE.Vector3();
+  	cameraDir_n.copy(cameraDir);	
+  	cameraDir_n.normalize();
+  	
+  
+  	var quaternion_y = new THREE.Quaternion(); 
+  	quaternion_y.setFromUnitVectors(new THREE.Vector3(0,0,-1), (new THREE.Vector3(cameraDir_n.x, 0, cameraDir_n.z)).normalize());
+  	var horizontal_rot = new THREE.Vector3(); 
+  	horizontal_rot = new THREE.Vector3(0,0,-1);
+  	horizontal_rot.applyQuaternion(quaternion_y);
+  	
+  	var quaternion_tilt = new THREE.Quaternion();
+  	quaternion_tilt.setFromUnitVectors(horizontal_rot, cameraDir_n);
+  	
+  	var shift = cameraDir_n.dot(new THREE.Vector3(_activeObjX, _activeObjY, _activeObjZ));
+  	
+  	
+  	var scaling = (cameraDir.length())*Math.tan(22.5* Math.PI / 180) / (this.divHeight /2); 
+  	var unrotated = new THREE.Vector3(_relX * scaling, -_relY * scaling, 0); 
+  	var rotated = unrotated.applyQuaternion(quaternion_y);
+  	
+  	rotated = rotated.applyQuaternion(quaternion_tilt);
+  	
+  	var newPos_vector = new THREE.Vector3(rotated.x, rotated.y, rotated.z );
+
+  	var newPos = new THREE.Vector3;
+  	newPos.addVectors(cameraDir, newPos_vector);
+  	newPos.normalize();
+
+  	newPos.multiplyScalar(Math.sqrt(Math.pow((cameraDir.length() + shift),2) + Math.pow(newPos_vector.length() * (cameraDir.length() + shift) / cameraDir.length(),2)));
+  	newPos.addVectors(newPos, cameraDir.negate());
+
+
+  	position[0] = newPos.x;
+  	position[1] = newPos.y;
+  	position[2] = newPos.z;
+  }
+
+  return position;
+
+};
+
+/**
+ * Loads a scene into the room viewer.
+ * @param {String} _roomXML Content of a room xml.
+ */
+Scene.prototype.load = function() {
+ 	 var parser = new DOMParser();
+   this.setAmbientLight(0xeeeeee);
+   this.setDirectionalLight();
+ 	 this.scene.add(this.camera);
+	 this.buildRectangularElement();
+};
+
+/**
+ *Renders the scene.
+ */
+Scene.prototype.render = function() {
+	this.renderer.render(this.scene, this.camera);
+};
+
+Scene.prototype.buildRectangularElement = function() {
+	this.width = this.pageConfig.roomMeasurements[0];
+	this.height = this.pageConfig.roomMeasurements[1];
+	this.depth = this.pageConfig.roomMeasurements[2];
+
+	var carpet = THREE.ImageUtils.loadTexture('lib/webmushra/spatial/resources/carpet.png');

```

Comment:
these resources would need to go somewhere else since we compress every js in `/lib`for a release js.